### PR TITLE
Use Devise current_publisher method

### PR DESCRIPTION
**Pull request by cesidio**

### CODE REVIEW

**Comment by martin-bangoura** [comment]
Review app deployed to <https://teaching-vacancies-review-pr-3064.london.cloudapps.digital>

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -81,7 +81,7 @@ def recaptcha_is_invalid?(model)
   end
 
   def request_event
-    RequestEvent.new(request, response, session, current_jobseeker, current_publisher_oid)
+    RequestEvent.new(request, response, session, current_jobseeker, current_publisher&.oid)
```
Comment (cpjmcquillan) [inline app/controllers/application_controller.rb:84]: Can we refactor the `RequestEvent` to take the current publisher the same way it takes the current jobseeker?

File: spec/components/shared/navbar_component_spec.rb [hunk 2]
```diff
@@ -5,6 +5,7 @@
 
   before do
     allow(controller).to receive(:jobseeker_signed_in?).and_return(true)
+    allow(controller).to receive(:publisher_signed_in?).and_return(false)
```
Comment (cpjmcquillan) [inline spec/components/shared/navbar_component_spec.rb:8]: Why is this needed?

File: spec/system/hiring_staff_have_to_accept_terms_spec.rb [hunk 3]
```diff
@@ -66,6 +67,10 @@
   end
 
   context "the user has accepted the terms and conditions" do
+    before do
+      stub_publishers_auth(urn: school.urn, oid: oid)
```
Comment (cpjmcquillan) [inline spec/system/hiring_staff_have_to_accept_terms_spec.rb:71]: Now we're using Devise, do we need to have this helper method at all? Could we just use devise `login_as`?

**Review by cpjmcquillan — APPROVED** [review]
Couple of comments that you might want to address

File: spec/system/hiring_staff_have_to_accept_terms_spec.rb (hunk 3 above)
Comment (cesidio) [reply spec/system/hiring_staff_have_to_accept_terms_spec.rb:71]: we still need to associate a publisher with an organisation, that is the next refactoring that I am doing

File: spec/components/shared/navbar_component_spec.rb (hunk 2 above)
Comment (cesidio) [reply spec/components/shared/navbar_component_spec.rb:8]: because otherwise we have errors with devise as it sets things that are not accessible from view components

File: spec/system/hiring_staff_have_to_accept_terms_spec.rb (hunk 3 above)
Comment (cpjmcquillan) [reply spec/system/hiring_staff_have_to_accept_terms_spec.rb:71]: Yeah I just realised that a second ago on my branch 😂
